### PR TITLE
[KAN-506] lambda function sending results to svc invoice

### PR DESCRIPTION
**Pull request by RyanKeepRunning**
https://aussiecommerce.atlassian.net/browse/KAN-506?atlOrigin=eyJpIjoiZmI1ZGY0Zjk5YmYyNDg1YmIyMjcxMTg0OTU2ZDMzZTAiLCJwIjoiaiJ9

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
```
Comment (marhalpert) [inline tsconfig.json:3]: Maybe good to clean up this file a bit

File: src/invoiceService/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      Authorization: `Bearer ${token}`
+    }
+  };
+
+  const destination = `${API_HOST}/api/invoices/flights/${vendorDomain}`;
+
+  console.log(`Sending processed data of ${vendorDomain} to svc-invoice`);
+  const response = await fetch(destination, options);
+
+  if (!response.ok) {
```
Comment (mynameisrufus) [inline src/invoiceService/index.ts:30]: easier to read with guard clauses

```ts
if (response.ok) {
  return
}

if (response.status === 400) {
 throw Error
}

throw Error

**Comment by mynameisrufus** [comment]
Needs eslint

**Comment by RyanKeepRunning** [comment]
> Needs eslint

I think it's already included? (.eslintrc.json)

File: src/authService/index.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+const CACHE_TIMEOUT = 3600;
+
+const CACHE_KEY = "access-token";
+
+const cache = new LRU({ maxAge: CACHE_TIMEOUT });
+
+export async function getToken(): Promise<string> {
+  console.log(`Retrieving access token`);
+  const token = cache.get(CACHE_KEY) as string;
```
Comment (innomatics) [inline src/authService/index.ts:16]: I'm not sure there is any point to cache the access token.  We only use it once and the lru cache won't persist between lambda invocations

Comment (innomatics) [reply src/authService/index.ts:16]: It's also a low use function so not need to set up a distributed cache

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        .promise();
+    }
+  }
+};
+
+const getVendorDomain = (srcKey: string): string => {
+  const vendorDomainMatches = srcKey.match(/\/([a-z\.]+)\//);
+  if (!vendorDomainMatches || vendorDomainMatches.length !== 2) {
+    throw new VendorValidationError(
+      "Could not determin vendorDomain from object key"
```
Comment (innomatics) [inline src/index.ts:111]:
```suggestion
      "Could not determine vendorDomain from object key"
```

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    );
+  }
+  const vendorDomain = vendorDomainMatches[1];
+  if (!vendorDomain.split(",").includes(vendorDomain)) {
```
Comment (innomatics) [inline src/index.ts:115]: This doesn't make sense to me - I think you should be using the whitelist env var here

Comment (RyanKeepRunning) [reply src/index.ts:115]: I was about to put whitelist here but then totally forgot about it! :(

**Pull request merged**